File: Algorithms/946_Validate_Stack_Sequences/solution.go
func validateStackSequences(pushed []int, popped []int) bool {
    stk := make([]int, 0, 1000)
    i := 0

    for _, element := range pushed {
        stk = append(stk, element)
        for len(stk) > 0 && stk[len(stk)-1] == popped[i] {
            stk = stk[:len(stk)-1]
            i++
        }
    }

    return len(stk) == 0
}

File: Algorithms/946_Validate_Stack_Sequences/solution.rb
# @param {Integer[]} pushed
# @param {Integer[]} popped
# @return {Boolean}
def validate_stack_sequences(pushed, popped)
  stk, i = [], 0
  pushed.each do |element|
    stk << element
    while !stk.empty? && stk[-1] == popped[i]
      stk.pop
      i += 1
    end
  end
  stk.empty?
end